core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -192,6 +195,18 @@ public Table create() {
 
       String baseLocation = location != null ? location : defaultWarehouseLocation(identifier);
       tableProperties.putAll(tableOverrideProperties());
+
+      if (Boolean.parseBoolean(
+          tableProperties.get(TableProperties.LOCATION_CONFLICT_DETECTION_ENABLED))) {
+        boolean conflictLocationDetected =
+            (ops.io() instanceof SupportsPrefixOperations)
+                ? !prefixEmpty((SupportsPrefixOperations) ops.io(), baseLocation)
+                : ops.io().newInputFile(baseLocation).exists();
```

Comment:
I think we should still try to get consistent behavior and try to list in all cases, even if IO is not SupportsPrefixIO. (ResolvingFileIO does not implement that, right?)  Otherwise it may be surprise to user.

Can we fallback to using File and call listStatus, like we do today in DeleteOrphanFilesSparkAction, in case FileIO does not support it).  We can probably load this differently depending on catalog type, ie, have HiveCatalog load File using its own HadoopConf, whereas others will take their default conf.

Not sure anyone has any thoughts, @RussellSpitzer ?

Reply:
thank you for the suggestion @szehon-ho . I added another check to see if io is instance of hadoop configurable, and then create a filesystem to see if directory is empty by listing recursively. 

There's ongoing PR #8334 trying to add prefix based operation in resolveFileIO. But I added test for both hadoop fileIO and resolving fileIO for each catalog just to make sure tests get passing regardless.